salt/metalk8s/solutions/available.sls

```diff
@@ -0,0 +1,120 @@
+#
+# Management of Solution archives, mount/unmount and expose container images
+#
+# Relies on the `archives` key from /etc/metalk8s/solutions.yaml
+#
+
+{%- from "metalk8s/map.jinja" import repo with context %}
+
+include:
+  - metalk8s.repo.installed
+
+{%- macro extract_info(archive_path) %}
+  
+  {{ machine_name }},{{ display_name }},{{ mount_path }}
+{%- endmacro %}
+
+{%- if '_errors' in pillar.metalk8s.solutions.config %}
+Cannot proceed with mounting of Solution archives:
+  test.fail_without_changes:
+    - comment: "Errors: {{ pillar.metalk8s.solutions.config._errors | join('; ') }}"
+
+{%- else %}
+  {#- Mount configured #}
+  {%- set configured = pillar.metalk8s.solutions.config.archives %}
+  {%- for archive_path in configured %}
+    {%- set solution = salt['metalk8s.archive_info_from_iso'](archive_path) %}
+    {%- set lower_name = solution.name | replace(' ', '-') | lower %}
+    {%- set machine_name = lower_name ~ '-' ~ solution.version %}
+    {%- set display_name = solution.name ~ ' ' ~ solution.version %}
+    {%- set mount_path = "/srv/scality/" ~ machine_name -%}
+
+{# Mount the archive #}
+Mountpoint for Solution {{ display_name }} exists:
+  file.directory:
+    - name: {{ mount_path }}
+    - makedirs: true
+
+Archive of Solution {{ display_name }} is mounted at {{ mount_path }}:
+  mount.mounted:
+    - name: {{ mount_path }}
+    - device: {{ archive_path }}
+    - fstype: iso9660
+    - mkmnt: false
+    - opts:
+      - ro
+      - nofail
+    - persist: true
+    - match_on:
+      - name
+    - require:
+      - file: Mountpoint for Solution {{ display_name }} exists
+
+{# Validate the archive contents
+   TODO: This should be moved before mounting the solution's ISO, using some
+   custom module #}
+Product information for Solution {{ display_name }} exists:
```

Comment:
If you try to make 2 different version of the same solution at the same time you will have conflict here as you will have 2 states with the same name.

You should change the display_name to also contains the version or add something unique for each state so "display_name" + "version" or "mount_path" but not "display_name" only

Reply:
See L.29: `{% set display_name = solution.name ~ '-' ~ solution.version %}`

Reply:
Oh yes MB through it was the `display_name` from `list_available` salt modules